### PR TITLE
Feat/LF-76 activate/deactivate all 

### PR DESCRIPTION
**Pull request by jeffersondarcy**
Buttons to activate and deactivate all switches in Marketing Notifications

### CODE REVIEW

File: src/components/notifications/preferences/ToggleAll.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        base: 340,
+        lg: '100%',
+    });
+
+    // use isEnabled=true to check if all preferences are enabled and isEnabled=false to check if all preferences are disabled
+    const isAllEnabledOrDisabled = (isEnabled: boolean = true) => {
+        return Object.keys(notificationCategories).every((category: string) =>
+            channels.every((channel: string) => {
+                if (!userPreferences.hasOwnProperty(category) || !userPreferences[category].hasOwnProperty(channel)) return true;
+                console.log(category, channel, userPreferences[category][channel]);
```
Comment (Jonasdoubleyou) [inline src/components/notifications/preferences/ToggleAll.tsx:33]: Debug logging? Either remove or add some context to the logline

File: src/components/notifications/preferences/ToggleAll.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import { NotificationPreferencesContext } from '../../../pages/notification/NotficationControlPanel';
+import { getAllPreferencesInCategorySetToValue } from '../../../helper/notification-helper';
+
+type PrefProps = {
+    notificationCategories: NotificationCategories;
+};
+
+export const ToggleAll: React.FC<PrefProps> = ({ notificationCategories }) => {
+    const { userPreferences, channels, updateUserPreferences } = useContext(NotificationPreferencesContext);
+    const allEnabled = useRef(false);
```
Comment (Jonasdoubleyou) [inline src/components/notifications/preferences/ToggleAll.tsx:14]: Why is this a ref? 

const allEnabled = useMemo(isAllEnables, [...]);

Comment (jeffersondarcy) [reply src/components/notifications/preferences/ToggleAll.tsx:14]: this value is recalculated in useEffect every time any switch is changed and is set true when all switches are enabled.  It's used as state, but without rerendering. Do you think useMemo is better suited for this?

Comment (Jonasdoubleyou) [reply src/components/notifications/preferences/ToggleAll.tsx:14]: > It's used as state, but without rerendering
Yes, but you actually need to rerender as `allEnabled.current` is used as a prop. This only works as the component is rerendered anyways whenever `allEnabled` is changed. That's quite dangerous.

Actually, `allEnabled` is not a state itself, but can be derived from another state - `userPreferences` - that's exactly what useMemo is good for - deriving complex calculations from state.

Comment (jeffersondarcy) [reply src/components/notifications/preferences/ToggleAll.tsx:14]: okay, I will try to incorporate this - I don't have experience with useMemo. can you explain the danger - the value is recalculated in useEffect always prior to setting state&rerender

File: src/components/notifications/preferences/ToggleAll.tsx (hunk 1 above)
Comment (jeffersondarcy) [reply src/components/notifications/preferences/ToggleAll.tsx]: removed

File: src/components/notifications/preferences/ToggleAll.tsx (hunk 2 above)
Comment (Jonasdoubleyou) [reply src/components/notifications/preferences/ToggleAll.tsx:16]: Well you can write to allEnabled.current, and the UI might not rerender. So the "state" can diverge from what the user sees.

Comment (jeffersondarcy) [reply src/components/notifications/preferences/ToggleAll.tsx:16]: done

Comment (jeffersondarcy) [reply src/components/notifications/preferences/ToggleAll.tsx]: sorry, forgot to push. now is updated

**Review by Jonasdoubleyou — APPROVED** [review]

**Pull request merged**